Framework/Algorithms/src/ChangePulsetime.cpp

```diff
@@ -38,41 +36,27 @@ void ChangePulsetime::init() {
 /** Execute the algorithm.
  */
 void ChangePulsetime::exec() {
-  EventWorkspace_const_sptr in_ws = getProperty("InputWorkspace");
+  EventWorkspace_sptr in_ws;
+  SpectrumIndexSet indexSet(0);
+
+  std::tie(in_ws, indexSet) = std::tuple<EventWorkspace_sptr, SpectrumIndexSet>(
```

Comment:
There does not seem to be a nice solution to this. The main issue is that we would need to convert to a tuple of references. See also here https://stackoverflow.com/questions/26276943/make-custom-type-tie-able-compatible-with-stdtie.

Given that, it seem to be a bit pointless to explicitly convert something to a tuple, just to use `tie` to unpack it right away. I think we should consider just doing to separate calls, i.e.,

```cpp
EventWorkspace_const_sptr in_ws = getProperty("InputWorkspace");
SpectrumIndexSet indexSet = getProperty("InputWorkspace");
```

Or maybe someone has a better solution.

In any case, I think you should bring this issue back into the design document, so other people can have a look at this.